### PR TITLE
Fetch full PR branch

### LINKED ISSUE #756: Project Checks fail when PR branch is long
I have a medium size (by my standard) PR and the `Project Checks` DCO check fails because it doesn't have enough commits to see the end.

```
  fatal: bad revision 'c101a5ec56fef7bf18021e895d8a861edd117c64^..c101a5ec56fef7bf18021e895d8a861edd117c64'
   * c101a5e "..." ... FAIL
    - FAIL - errored with: exit status 128
```

There are other ways to write the CI so that it would work.

### PR DESCRIPTION
**Pull request by jsoref**
Closes #756

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -20,6 +20,14 @@ jobs:
       with:
         path: src/github.com/containerd/nerdctl
         fetch-depth: 25
```
Comment (AkihiroSuda) [inline .github/workflows/test.yml:22]: Can we just increase fetch-depth here?

Comment (jsoref) [reply .github/workflows/test.yml:22]: You could pick any arbitrary number, it'll always eventually be wrong.

I personally use a different approach for my own workflows, but it's more complicated than what you have (which is broken) or what I'm proposing (which will solve the PR case, but wouldn't technically solve a Push event with a similarly long set of commits.

Comment (AkihiroSuda) [reply .github/workflows/test.yml:22]: > You could pick any arbitrary number, it'll always eventually be wrong.

100 should be enough for everyone.

If you really need to fetch all commits you can just remove `fetch-depth` line, but I don't see necessity of doing that.

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]
https://github.com/containerd/nerdctl/pull/757#discussion_r795384162

**Comment by jsoref** [comment]
I accidentally broke the link to this branch, closing -- will try to reopen.

**Pull request closed without merging**